apps/astro/src/load-query.ts

```diff
@@ -0,0 +1,20 @@
+import {type QueryParams} from 'sanity'
+import {sanityClient} from 'sanity:client'
+
+const isDraftMode = import.meta.env.SANITY_VISUAL_EDITING_ENABLED === 'true'
+const token = import.meta.env.SANITY_API_READ_TOKEN
+
+export function loadQuery<QueryResponse>({query, params}: {query: string; params?: QueryParams}) {
+  if (isDraftMode && !token) {
+    throw new Error('The `SANITY_API_READ_TOKEN` environment variable is required in Draft Mode.')
+  }
+
+  return sanityClient.fetch<QueryResponse>(query, params ?? {}, {
+    filterResponse: false,
+    perspective: isDraftMode ? 'previewDrafts' : 'published',
+    resultSourceMap: isDraftMode ? 'withKeyArraySelector' : false,
+    stega: isDraftMode,
+    ...(isDraftMode ? {token} : {}),
+    useCdn: !isDraftMode,
+  })
```

Comment:
I would change this to follow the same pattern as `@sanity/react-loader`'s and `@sanity/svelte-loader`'s implementations:

```suggestion

  const perspective = isDraftMode ? 'previewDrafts' : 'published'
  const {result, resultSourceMap} = await sanityClient.fetch<QueryResponse>(query, params ?? {}, {
    filterResponse: false,
    perspective,
    resultSourceMap: isDraftMode ? 'withKeyArraySelector' : false,
    stega: isDraftMode,
    ...(isDraftMode ? {token} : {}),
    useCdn: !isDraftMode,
  })
  return {data: result, sourceMap: resultSourceMap, perspective}
```

Reply:
Good idea!